.snippets/code/vs-ethereum/tx-fees-block-2100.md

```diff
@@ -0,0 +1,106 @@
+
+// Endpoint to retrieve the latest block
+const endpointBlock = "http://127.0.0.1:8080/blocks/head";
+// Endpoint to retrieve the latest nextFeeMultiplier
+const endpointPallet = "http://127.0.0.1:8080/pallets/transaction-payment/storage/nextFeeMultiplier?at=";
+// Endpoint to retrieve the node client's information
+const endpointNodeVersion = "http://127.0.0.1:8080/node/version";
+
+async function main() {
+  try {
+    // Create a variable to sum the transaction fees in the whole block
+    let totalFees = 0n;
+
+    // Find which Moonbeam network the Sidecar is pointing to
+    const response_client = await axios.get(endpointNodeVersion);
+    const network = response_client.data.clientImplName;
+
+    // Retrieve the block from the Sidecar endpoint
+    const response_block = await axios.get(endpointBlock);
+    // Retrieve the block height of the current block
+    console.log("Block Height: " + response_block.data.number);
+
+    // Find the block's nextFeeMultiplier
+    const response_pallet = await axios.get(endpointPallet + response_block.data.number);
+
+    // Iterate through all extrinsics in the block
+    response_block.data.extrinsics.forEach((extrinsic) => {
+      // Create an object to store transaction information
+      let transactionData = new Object();
+      // Set the network field
+      transactionData["network"] = network;
+
+      // Filter for Ethereum Transfers
+      if (extrinsic.method.pallet === "ethereum" && extrinsic.method.method === "transact") {
+        // Iterate through the events to get non type specific parameters
+        extrinsic.events.forEach((event) => {
+          if (event.method.pallet === "ethereum" && event.method.method === "Executed") {
+            // Get Transaction Hash
+            transactionData["hash"] = event.data[2];
+          }
+          if (event.method.pallet === "system" && event.method.method === "ExtrinsicSuccess") {
+            // Add correction weight if needed to Transaction Weight!
+            transactionData["weight"] = BigInt(event.data[0].weight);
+          }
+        });
+
+        // Get the transaction type and type specific parameters and compute the transaction fee
+        if (extrinsic.args.transaction.legacy) {
+          transactionData["txType"] = "legacy";
+          transactionData["gasPrice"] = BigInt(extrinsic.args.transaction.legacy.gasPrice);
+          transactionData["txFee"] = (transactionData["gasPrice"] * transactionData["weight"]) / 25000n;
+        } else if (extrinsic.args.transaction.eip1559) {
+          transactionData["txType"] = "eip1599";
+          transactionData["maxFeePerGas"] = BigInt(extrinsic.args.transaction.eip1559.maxFeePerGas);
+          transactionData["maxPriorityFeePerGas"] = BigInt(extrinsic.args.transaction.eip1559.maxPriorityFeePerGas);
+          transactionData["baseFee"] = BigInt(response_pallet.data.value) * 1250000000 / BigInt("1000000000000000000")
```

Comment:
Not really sure if what I am doing is correct, but after I used `refTime` to jump over first error then I received this one: `TypeError: Cannot mix BigInt and other types, use explicit conversions`

Reply:
I think here it just needs to be `1250000000n` instead